### PR TITLE
LF-3091 The user is able to create a duplicate of the crop that already exists in the farm if he lefts fields : "Genus" & "Species" blank

### PR DESCRIPTION
**Pull request by kathyavini**
**Description**

This PR adds a uniqueness constraint to adding a crop in the frontend, so that the Add Crop form cannot be submitted if that combination of Crop, Genus, and Species already exists on the farm.

Additionally, this constraint has been added to the API, so that the submission will fail also on the API endpoint if that is accessed directly.

**This ticket has a language component for the validation error displayed when a Crop is not unique.**

Jira link: https://lite-farm.atlassian.net/browse/LF-3091

**Type of change**

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**How Has This Been Tested?**

- [ ] Passes test case
- [x] UI components visually reviewed on desktop view
- [x] UI components visually reviewed on mobile view
- [ ] Other (please explain)

**Checklist:**

- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] The precommit and linting ran successfully
- [x] I have added or updated language tags for text that's part of the UI
- [x] I have added "MISSING" for all new language tags to languages I don't speak
- [ ] I have added [the GNU General Public License](https://lite-farm.atlassian.net/l/cp/BT0Dd7WW) to all new files

### CODE REVIEW

**Review by Pazzolo — APPROVED** [review]

**Comment by kathyavini** [comment]
Sorry @Pazzolo and @Duncan-Brain! I didn't look closely enough to see that the database PR was raised against this branch and not integration.

Just merged the database change and invalidated the review, but now the whole functionality is ready for review and merge. ❤️ Thanks!

File: packages/webapp/src/components/AddNewCrop/index.jsx [hunk 1]
```diff
@@ -81,6 +85,40 @@ export default function PureAddNewCrop({
   const cropImageUrlRegister = register(CROP_PHOTO_URL, { required: true });
   const crop_photo_url = watch(CROP_PHOTO_URL);
 
+  const farmCropsSelector = useSelector(cropsOnMyFarmSelector);
```
Comment (SayakaOno) [inline packages/webapp/src/components/AddNewCrop/index.jsx:88]: I think this should be `farmCrops` not selector.

File: packages/api/src/controllers/cropController.js [hunk 2]
```diff
@@ -68,7 +68,20 @@ const cropController = {
     return async (req, res) => {
       const trx = await transaction.start(Model.knex());
       try {
-        const { crop, variety } = req.body;
+        const { crop, variety, farm_id } = req.body;
+        const isDuplicateCrop = await CropModel.query().findOne({
```
Comment (SayakaOno) [inline packages/api/src/controllers/cropController.js:72]: I think the variable name should be `duplicateCrop` or something.

**Review by SayakaOno — COMMENTED** [review]
Great job!
I just have small suggestions on variable names.

**Comment by SayakaOno** [comment]
Probably I missed something, but will the user be able to create a crop named "Strawberry"?

File: packages/webapp/src/components/AddNewCrop/index.jsx (hunk 1 above)
Comment (kathyavini) [reply packages/webapp/src/components/AddNewCrop/index.jsx:88]: Yes it should, thank you 🙂

**Comment by kathyavini** [comment]
@SayakaOno YES! The ticket reads "If a crop_type with the same name exists with that farm_id" which only applies to created crops.

But you're saying we should not be able to replicate an existing crop catalogue name either 🤔 That makes total sense to me! But no, these changes wouldn't do that.

I think we also need the validation you describe. For the patch??!

**Comment by SayakaOno** [comment]
@kathyavini 
I remember the example Kevin used to describe the duplicated crop issue was strawberry.
Yes! For the patch!!!

**Comment by kathyavini** [comment]
@SayakaOno he was talking about the Varietal common name when he gave the Strawberry example! But I think what you're saying would absolutely be in the spirit of what he wants for crop, so I will tag him on the Jira ticket to add this in!

**Comment by SayakaOno** [comment]
Ah okay! Thank you!!

**Review by SayakaOno — APPROVED** [review]

**Pull request merged**